codecarbon/core/powermetrics.py

```diff
@@ -0,0 +1,136 @@
+import os
+import re
+import shutil
+import subprocess
+import sys
+from typing import Dict
+
+import numpy as np
+
+from codecarbon.core.util import detect_cpu_model
+from codecarbon.external.logger import logger
+
+
+def is_powermetrics_available():
+    try:
+        ApplePowermetrics()
+        return True
+    except Exception as e:
+        logger.debug(
```

Comment:
Do you think it's possible to detect that powermetrics exist but the user is not root to tell him how to add right ?
Like we do for RAPL: https://github.com/mlco2/codecarbon/blob/master/codecarbon/core/cpu.py#L218

Reply:
I will have a look into this